creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -1004,6 +927,103 @@ class SnapbackSM {
+   * Purpose for the if/else case is that if the current node is a primary, issue reconfig or sync requests.
+   * Else, if the current node is a secondary, only issue reconfig requests.
+   *
+   * @notice this will issue sync to healthy secondary and update replica set away from unhealthy secondary
+   */
+  async aggregateReconfigAndPotentialSyncOps (nodeUsers, unhealthyPeers) {
+    const requiredUpdateReplicaSetOps = []
+    const potentialSyncRequests = []
+
+    for (const nodeUser of nodeUsers) {
+      const { primary, secondary1, secondary2, primarySpID, secondary1SpID, secondary2SpID } = nodeUser
+
+      let unhealthyReplicas = []
+
+      /**
+       * If this node is primary for user, check both secondaries for health
+       * Enqueue SyncRequests against healthy secondaries, and enqueue UpdateReplicaSetOps against unhealthy secondaries
+       */
+      let replicaSetNodesToObserve = [
+        { endpoint: secondary1, spId: secondary1SpID },
+        { endpoint: secondary2, spId: secondary2SpID }
+      ]
+
+      if (primary === this.endpoint) {
+        // filter out false-y values to account for incomplete replica sets
+        replicaSetNodesToObserve = replicaSetNodesToObserve.filter(entry => entry.endpoint && entry.spId)
+
+        /**
+         * If either secondary is in `unhealthyPeers` list, add it to `unhealthyReplicas` list
+         */
+        const userSecondarySyncMetrics = await this._computeUserSecondarySyncSuccessRates(nodeUser, [secondary1, secondary2])
+        for (const secondary of replicaSetNodesToObserve) {
+          const secUserSyncSuccessRate = userSecondarySyncMetrics[secondary.endpoint]['SuccessRate']
+          if (secUserSyncSuccessRate < this.MinimumSecondaryUserSyncSuccessPercent ||
+            unhealthyPeers.has(secondary.endpoint) ||
+            this.peerSetManager.endpointToSPIdMap[secondary.endpoint] !== secondary.spId) {
+            unhealthyReplicas.push(secondary.endpoint)
+          } else {
+            potentialSyncRequests.push({ ...nodeUser, endpoint: secondary.endpoint })
+          }
+        }
+
+        /**
+         * If any unhealthy replicas found for user, enqueue an updateReplicaSetOp for later processing
+         */
+        if (unhealthyReplicas.length > 0) {
+          requiredUpdateReplicaSetOps.push({ ...nodeUser, unhealthyReplicas })
+        }
+
+        /**
+         * If this node is secondary for user, check both secondaries for health and enqueue SyncRequests against healthy secondaries
+         * Ignore unhealthy secondaries for now
+         */
+      } else {
+        // filter out false-y values to account for incomplete replica sets and filter out the
+        // the self node
+        replicaSetNodesToObserve = [{ endpoint: primary, spId: primarySpID }, ...replicaSetNodesToObserve]
+        replicaSetNodesToObserve = replicaSetNodesToObserve.filter(entry => {
+          return entry.endpoint && entry.spId && entry.endpoint !== this.endpoint
+        })
```

Comment:
no need to change but i think keeping the boolean and != endpoint checks separate is more readable

Reply:
hmm ok, how would you restructure this?

Reply:
reverting back to old code

Reply:
personally, I think the readability is ok here when combining the two conditions. im gonna leave it as is if that's alright

Reply:
yep, its nbd